app/scripts/controllers/transactions/index.js

```diff
@@ -337,9 +337,10 @@ export default class TransactionController extends EventEmitter {
    *
    * @param txParams
    * @param origin
+   * @param transactionType
    * @returns {txMeta}
    */
-  async addUnapprovedTransaction(txParams, origin) {
+  async addUnapprovedTransaction(txParams, origin, transactionType) {
```

Comment:
To make sure that this new parameter is not accidentally misused, could we validate this third parameter by checking that, `if (transactionType !== undefined)` then `transactionType` should equal either `TRANSACTION_TYPES.SWAP_APPROVAL` or `TRANSACTION_TYPES.SWAP`, and otherwise an error should be thrown